BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/single_file/single_file.rb

```diff
@@ -97,4 +97,27 @@ def cache_compute_md5sum
     cb_error("Can't compute MD5 for file #{self.name}") if md5.blank?
     md5
   end
+
+  def method_missing(name, *args)
+    return super unless name.to_s.match(/^size_allows_viewing_(.+)/)
+    size_limit   = Regexp.last_match[1].to_i
+    define_singleton_method (name) do 
+      size_allows_viewing?(size_limit) 
+    end
+    self.send(name)
+  end
+  
+  # Check if a file respect a size_limit in order to be viewable
+  # this method is used in the :if of the has_viewer used in specific 
+  # Userfile model
+  def size_allows_viewing?(size_limit=400_000)
+    userfile_errors = []
+    if self.size.blank?
+      userfile_errors.push("No size available for this file")  
+    else
+      userfile_errors.push("File is too large to be viewable (> #{size_limit.to_s} kB) ") if self.size > size_limit
```

Comment:
The message is wrong, it says `size_limit kb` but the limit is in bytes, as implemented.

You have to decide what the meaning of the 400_000 is. Is it bytes or kilobytes?

Reply:
You can also just change the message, by replacing `kB` with `bytes`.

Reply:
However, if you do that, then when a programmer wants to specify a limit of 5 gigabytes for viewing a file, they'll have to do 

```
has_viewer ...., :if => :size_allows_viewing_5_000_000_000
```

which is quite long and has a lot of zeros.